diff --git a/plugin/torch/_build_config.py b/plugin/torch/_build_config.py
index 7b009d0..16a23ef 100644
--- a/plugin/torch/_build_config.py
+++ b/plugin/torch/_build_config.py
@@ -170,5 +170,6 @@ def get_device_config(adaptor_flag):
     elif adaptor_flag == "-DUSE_DU_ADAPTOR":
-        include_dirs += ["${CUDA_PATH}/include"]
-        library_dirs += ["${CUDA_PATH}/lib64"]
-        libs += ["cuda", "cudart", "c10_cuda", "torch_cuda"]
+        cuda_path = os.environ["CUDA_PATH"]
+        include_dirs += [os.path.join(cuda_path, "include")]
+        library_dirs += [os.path.join(cuda_path, "lib64")]
+        libs += ["cuda", "cudart", "c10_hip", "torch_hip"]
     elif adaptor_flag == "-DUSE_KUNLUNXIN_ADAPTOR":
diff --git a/plugin/torch/flagcx/include/event_flagcx.hpp b/plugin/torch/flagcx/include/event_flagcx.hpp
index f7bda96..25f8792 100644
--- a/plugin/torch/flagcx/include/event_flagcx.hpp
+++ b/plugin/torch/flagcx/include/event_flagcx.hpp
@@ -28,4 +28,4 @@
 #elif USE_DU_ADAPTOR
-#include <ATen/cuda/CUDAEvent.h>
-#include <cuda_runtime.h>
+#include <ATen/hip/HIPEvent.h>
+#include <hip/hip_runtime.h>
 #elif USE_KUNLUNXIN_ADAPTOR
@@ -226,8 +226,9 @@
 #elif USE_DU_ADAPTOR
 class flagcxDuEvent : public flagcxEvent {
 public:
-  flagcxDuEvent() { cudaEvent_ = at::cuda::CUDAEvent(cudaEventDisableTiming); }
+  flagcxDuEvent() { cudaEvent_ = at::cuda::CUDAEvent(hipEventDisableTiming); }
-
+
   void record(const int deviceId) override {
-    cudaEvent_.record(at::cuda::getCurrentCUDAStream(deviceId));
+    cudaEvent_.record(
+        at::hip::getCurrentHIPStreamMasqueradingAsCUDA(deviceId));
   }
@@ -235,4 +236,4 @@ public:
   void record(const flagcxStream_t &stream, const int deviceId) override {
-    cudaEvent_.record(
-        at::cuda::getStreamFromExternal(*(cudaStream_t *)stream, deviceId));
+    cudaEvent_.record(at::hip::getStreamFromExternalMasqueradingAsCUDA(
+        *(hipStream_t *)stream, deviceId));
   }
@@ -240,3 +241,4 @@ public:
   void block(const int deviceId) override {
-    cudaEvent_.block(at::cuda::getCurrentCUDAStream(deviceId));
+    cudaEvent_.block(
+        at::hip::getCurrentHIPStreamMasqueradingAsCUDA(deviceId));
   }
@@ -244,4 +246,4 @@ public:
   void block(const flagcxStream_t &stream, const int deviceId) override {
-    cudaEvent_.block(
-        at::cuda::getStreamFromExternal(*(cudaStream_t *)stream, deviceId));
+    cudaEvent_.block(at::hip::getStreamFromExternalMasqueradingAsCUDA(
+        *(hipStream_t *)stream, deviceId));
   }
diff --git a/plugin/torch/flagcx/include/stream_guard_flagcx.hpp b/plugin/torch/flagcx/include/stream_guard_flagcx.hpp
index cff547a..698f5cb 100644
--- a/plugin/torch/flagcx/include/stream_guard_flagcx.hpp
+++ b/plugin/torch/flagcx/include/stream_guard_flagcx.hpp
@@ -38,5 +38,5 @@
 #include <c10/core/impl/InlineStreamGuard.h>
-#include <c10/cuda/CUDAGuard.h>
-#include <c10/cuda/impl/CUDAGuardImpl.h>
-#include <cuda_runtime.h>
+#include <c10/hip/HIPGuard.h>
+#include <c10/hip/impl/HIPGuardImpl.h>
+#include <hip/hip_runtime.h>
 #elif USE_KUNLUNXIN_ADAPTOR
@@ -87,4 +87,4 @@ public:
 #elif USE_DU_ADAPTOR
-        guard_(
-            at::cuda::getStreamFromExternal(*(cudaStream_t *)stream, deviceId))
+        guard_(at::hip::getStreamFromExternal(*(hipStream_t *)stream,
+                                              deviceId))
 #elif USE_KUNLUNXIN_ADAPTOR
@@ -150,4 +150,4 @@ public:
 #elif USE_DU_ADAPTOR
-    guard_.reset_stream(
-        at::cuda::getStreamFromExternal(*(cudaStream_t *)stream, deviceId_));
+    guard_.reset_stream(at::hip::getStreamFromExternal(
+        *(hipStream_t *)stream, deviceId_));
 #elif USE_KUNLUNXIN_ADAPTOR
@@ -192,3 +192,3 @@ private:
 #elif USE_DU_ADAPTOR
-  c10::cuda::CUDAStreamGuard guard_;
+  c10::hip::HIPStreamGuard guard_;
 #elif USE_KUNLUNXIN_ADAPTOR
diff --git a/setup.py b/setup.py
index e5e191d..51f8100 100644
--- a/setup.py
+++ b/setup.py
@@ -159,3 +159,9 @@ if CppExtension is not None:
         include_dirs=include_dirs,
-        extra_compile_args={"cxx": [adaptor_flag, torch_flag]},
+        extra_compile_args={
+            "cxx": [
+                adaptor_flag,
+                torch_flag,
+                "-DC10_CUDA_NO_CMAKE_CONFIGURE_FILE",
+            ]
+        },
         extra_link_args=[],
